### ISSUE TITLE
[WebGL2] NPOT Texture should not fallback in webgl2

### ISSUE THREAD

**Issue by shawn0326** [mrdoob/three.js#15554]
##### Description of the problem

These fallback logic should be skipped in webgl2:
````javascript

// webgl2 skip this
_gl.texParameteri( textureType, _gl.TEXTURE_WRAP_S, _gl.CLAMP_TO_EDGE );
_gl.texParameteri( textureType, _gl.TEXTURE_WRAP_T, _gl.CLAMP_TO_EDGE );
_gl.texParameteri( textureType, _gl.TEXTURE_MAG_FILTER, filterFallback( texture.magFilter ) );
_gl.texParameteri( textureType, _gl.TEXTURE_MIN_FILTER, filterFallback( texture.minFilter ) );

// isPowerOfTwo should be (isPowerOfTwo || webgl2)
function textureNeedsGenerateMipmaps( texture, isPowerOfTwo ) {
	return texture.generateMipmaps && isPowerOfTwo &&
			texture.minFilter !== NearestFilter && texture.minFilter !== LinearFilter;
}
````

##### Three.js version

- [x ] Dev
- [ ] r100
- [ ] ...

##### Browser

- [x] All of them
- [ ] Chrome
- [ ] Firefox
- [ ] Internet Explorer

##### OS

- [x] All of them
- [ ] Windows
- [ ] macOS
- [ ] Linux
- [ ] Android
- [ ] iOS

**Comment by mrdoob** [comment]
Sounds good to me.

@Mugen87 Does this conflict with #15286?

**Comment by Mugen87** [comment]
Yes, since `textureNeedsPowerOfTwo()` always returns `false` with a WebGL 2 rendering context. So `textureNeedsGenerateMipmaps()` needs a fix as well as `setTextureParameters()`.But I think the following two lines also needs an enhancement.

https://github.com/mrdoob/three.js/blob/dev/src/renderers/webgl/WebGLTextures.js#L613
https://github.com/mrdoob/three.js/blob/dev/src/renderers/webgl/WebGLTextures.js#L673

**Comment by takahirox** [comment]
OK, I'll make PR to fix.

**Comment by takahirox** [comment]
I have difficulty naming variables in English. How var name `var varname = ( isPowerOfTwo || capabilities.isWebGL2 );` should be? `isValidSize`?

**Comment by Mugen87** [comment]
Maybe `needsPowerOfTwo`? It's already used in the file but maybe not in your relevant scope.

**Comment by WestLangley** [comment]
How about this?
```javascript
var supportsMips = ( isPowerOfTwo || capabilities.isWebGL2 );
```

**Comment by takahirox** [comment]
Thanks for naming. I made PR #15574
